src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,189 @@
+
+        userInput = in.nextLine();
+        String[] inputWords = userInput.split(" ");
+        String userCommand = inputWords[0];
+
+        try {
+            switch (userCommand) {
+            case "bye":
+                printLine();
+                System.out.println("Bye. Hope to see you again soon!");
+                printLine();
+                break;
+            case "list":
+                printLine();
+                System.out.println("Here are the tasks in your list:");
+                showList();
+                printLine();
+                handleCommand();
+                break;
+            case "done":
+                printLine();
+                System.out.println("Nice! I've marked this task as done:");
+                markTaskAsDone(inputWords[1]);
+                printLine();
+                handleCommand();
+                break;
+            case "deadline":
+                addDeadlineOrEventTask(inputWords, "deadline");
+                handleCommand();
+                break;
+            case "event":
+                addDeadlineOrEventTask(inputWords, "event");
+                handleCommand();
+                break;
+            case "todo":
+                addTodoTask(inputWords);
+                handleCommand();
+                break;
+            default:
+                throw new InvalidCommandException();
+
+            }
+        } catch (InvalidCommandException e) {
+            printLine();
+            System.out.println("OOPS! I'm sorry, but I don't know what that means! :(");
+            System.out.println("Available commands: deadline, todo, event, done, list, bye");
+            printLine();
+            handleCommand();
+        } catch (EmptyCommandArgumentException e) {
+            printLine();
+            System.out.println("OOPS! The description of deadline/event/todo cannot be empty! " +
+                    "Please follow this format:");
+            System.out.println("deadline <your task here> /by <your deadline time>");
+            System.out.println("event <your task here> /at <your event time period>");
+            System.out.println("todo <your task here>");
+            printLine();
+            handleCommand();
+        } catch (InvalidCommandSeparatorException e) {
+            printLine();
+            System.out.println("OOPS! The deadline/event description must be separated from " +
+                    "the time using '/by' or '/at'. Please follow this format:");
+            System.out.println("deadline <your task here> /by <your deadline time>");
+            System.out.println("event <your task here> /at <your event time period>");
+            printLine();
+            handleCommand();
+        }
+    }
+
+    public void addDeadlineOrEventTask(String[] inputWords, String type)
+            throws EmptyCommandArgumentException, InvalidCommandSeparatorException {
+        // Throw exception where command argument is empty
```

Comment:
Do see if comments are unnecessary are they can be removed with better variable naming. If they are not necessary do remove them.